core/services/workflows/engine.go

```diff
@@ -242,7 +242,13 @@ func (e *Engine) init(ctx context.Context) {
 	})
 
 	if retryErr != nil {
-		e.logger.Errorf("initialization failed: %s", retryErr)
+		// TODO ks-461
+
+		// ks-461 Option A:
+		logAndSendCustomMessage(e.logger, "initialization failed: %s", retryErr)
```

Comment:
I slightly prefer this option: this doesn't couple our logger + custom message implementation together so we can flexible diverge the two (logs + custom messages). It'll make things more verbose, but also more explicit.

Reply:
The tricky part with A is getting access to all the sugared values that have been added to the logger. Maybe the best path forward is to look for specific values in the drilled context? For tracing those labels will be valuable as well if added to the context.

Reply:
I've gone with the context approach: 
700a38aaa598423a70feb3c29e3061ee669884a5

Reply:
Ultimately went with the `with` pattern